packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -131,12 +140,20 @@ class HTTPCommunicationQueue {
       return cachedResult
     }
 
-    const promise = this.#createMultipartUpload(file, signal).then(async (result) => {
+    const promise = this.#createMultipartUpload(file, signal)
+
+    const abortPromise = () => {
+      promise.abort(signal.reason)
+      this.#cache.delete(file.data)
+    }
+    signal.addEventListener('abort', abortPromise, { once: true })
+    this.#cache.set(file.data, promise)
```

Comment:
I knew this wasn't introduced in this PR, but why are we setting `file.data` to cache? Aren't we setting a 5MB (or worse) buffer string into the cache? Why don't we use `file.id`?

Reply:
It's a WeakMap, the size of the data doesn't matter here – using a string would be wore actually, as it wouldn't be able to garbage collect it.

Reply:
> the size of the data doesn't matter here

I find that hard to believe. If someone sets the part size to 50MB and we do this for every request, I can't imagine this wouldn't show up if would do some performance profiling?

Reply:
Yet that's how `WeakMap` work, think of it like a pointer in C, it doesn't matter if the pointer points to a very big chunk of memory or a very small one.